enhancements/windows-containers/windows-machine-config-operator.md

```diff
@@ -0,0 +1,165 @@
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Provide workflows for adding Windows instances provisioned using the Machine
+  API to OpenShift clusters.
+* Perform all the required steps within the node for it to be an OpenShift
+  compute node
+
+### Non-Goals
+
+As part of this enhancement we do not plan to support:
+* Installing the container runtime in the Windows node
+* Upgrades (This will be addressed in a separate enhancement)
+  * Windows operating system upgrades
+  * Node management like k8s compute component upgrades, reboots and node
+    draining
+* OpenShift Builds
+* OpenShift DeploymentConfigs
+
+## Proposal
+
+The two main pieces that will allow us to achieve our goals is the Windows
+Machine Config operator (WMCO) and a Windows Machine Config Bootstrapper (WMCB)
+binary. The operator will collect information from the cluster and transfer it
+to the Windows node. The WMCB will then perform the necessary steps on the node
+for it to join the cluster. [WMCB](https://github.com/openshift/windows-machine-config-bootstrapper)
+has already been developed as part of the Windows Containers dev preview.
+
+### Justification
+
+The reason we are using a WMCB binary rather than a container image is
+that all Windows container images contains a Windows kernel and Red Hat has a
+policy to not ship 3rd party kernels for support reasons.
+
+### Design Details
+
+The operator image will be packaged with the following binaries:
+* Kubelet
+* CNI plugins
+* Hybrid-overlay
+* WMCB
+* Kube-proxy
+
+WMCO will be published to OperatorHub as a Red Hat operator. The way the cluster
+admin will enable Windows workloads on their cluster would be to first install
+WMCO. To express that they want to bring up Windows worker nodes, they will
+create a MachineSet where they reference a Windows OS image which has the
+Docker container runtime add-on enabled. The operator will watch for Machines
+objects with a special label that indicate that they map to Windows machines
+(TBD). Once the Machine object is in the "provisioned" state, the operator will
+connect to it transfer the binaries and perform the following steps for it to
+join the cluster as a worker node:
+* Transfer the required binaries
+* Configure the kubelet service using WMCB
+* Run the hybrid-overlay-node.exe binary
+* Configure the kubelet service for CNI using WMCB
+* Configure kube-proxy
+
+The operator expects the cluster admin to create a well know Secret in it's
+namespace with a private key that will be used to interact with the Windows
+instance. The operator will check for this secret during boot up time and create
+a user data Secret which the cluster admin can reference in the Windows
+MachineSet that they create.
+
+TBD: How do we get MCO to ignore Windows nodes?
+
+### Risks and Mitigations
+
+The main risk is that OVN Hybrid networking will not GA in 4.6. In that scenario
+an alternative networking solution like Calico will need to be explored.
+
```

Comment:
We are not targetting Bare Metal or RHV for 4.6. Please see the summary where I have called out all the targeted platforms.

Given that we are in the process of solving the vSphere networking issue, it would become a GA blocker bug that need not be called here.